cmd/osbuild-dnf-json-tests/main_test.go

```diff
@@ -80,12 +80,12 @@ func TestCrossArchDepsolve(t *testing.T) {
 							imgType, err := arch.GetImageType(imgTypeStr)
 							require.NoError(t, err)
 
-							buildPackages := imgType.BuildPackages()
-							_, _, err = rpm.Depsolve(buildPackages, []string{}, repos[archStr], distroStruct.ModulePlatformID(), archStr)
+							packages := imgType.PackageSets(blueprint.Blueprint{})
+
+							_, _, err = rpm.Depsolve(packages["build-packages"].Include, packages["build-packages"].Exclude, repos[archStr], distroStruct.ModulePlatformID(), archStr)
 							assert.NoError(t, err)
 
-							basePackagesInclude, basePackagesExclude := imgType.Packages(blueprint.Blueprint{})
-							_, _, err = rpm.Depsolve(basePackagesInclude, basePackagesExclude, repos[archStr], distroStruct.ModulePlatformID(), archStr)
+							_, _, err = rpm.Depsolve(packages["packages"].Include, packages["build-packages"].Exclude, repos[archStr], distroStruct.ModulePlatformID(), archStr)
```

Comment:
`packages["build-packages"].Exclude` is wrong on this line.

Maybe let's move `PackageSet` into `rpmmd` and make `rpmmd.Depsolve` taking `PackageSet` as an argument? This change would prevent this kind of mistakes (it happened in past, see #1148)
Maybe `rpm.Depsolve` should take `distro.PackageSet` as an argument?

Reply:
Good catch, and good suggestion! Done.